### ISSUE TITLE
Inconsistent results between Automatic1111 and Diffusers version IP-Adapter

### ISSUE THREAD

**Issue by KevinChen880723** [tencent-ailab/IP-Adapter#122]
Dear authors,

Thanks for open-sourcing such a great project, it helps me a lot!
I'm currently working on my research based on this project and want to deploy the trained weights to Automatic1111.
The results of diffusers are very good but I have encountered a noticeable degradation when deploying them on Automatic1111.
I am reaching out to kindly request your guidance or any hints that might help me resolve this inconsistency issue. Your assistance would be greatly appreciated. Thank you very much in advance!

I use the following code to save the diffusers weights:
```python
out_state_dict = dict()
out_state_dict["image_proj"] = image_proj_model.cpu().state_dict()
out_state_dict["ip_adapter"] = adapter_modules.cpu().state_dict()
torch.save(out_state_dict, 'webui-controlnet-ip-adapter.pth')
pipe.save_pretrained('saved/ip_adapter/', variant="fp16")
```
Subsequently, I employed `diffusers/scripts/convert_diffusers_to_original_stable_diffusion.py` script to convert the weights located under `saved/ip_adapter/` to the `*.safetensors` format:
```bash
python scripts/convert_diffusers_to_original_stable_diffusion.py --model_path saved/ip_adapter --checkpoint_path ../stable-diffusion-webui/models/Stable-diffusion/dreambooth_ipadapter.safetensors --half --use_safetensors
```

Once again, thank you for your contribution, and I look forward to any insights you may provide.

Sincerely,
Kevin

**Comment by xiaohu2015** [comment]
you can just use `webui-controlnet-ip-adapter.pth`, but don't convert to safetensors.

**Comment by KevinChen880723** [comment]
@xiaohu2015 
Thanks for your quick response!
Seems I didn't describe my problems correctly lol

In my specific use case, I'm not just fine-tuning the IP-Adapter part, but also the UNet part. Therefore, I save the modified UNet weights using the following command: pipe.save_pretrained('saved/ip_adapter/', variant="fp16"). It seems that the checkpoint format under the 'Stable Diffusion checkpoint' option in Automatic1111 must use safetensors, which is why I converted it.
Regarding the IP-Adapter weights, I didn't perform any conversions and simply left them as .pth files.

Thank you once more! Looking forward to any insights you may offer!

Sincerely,
Kevin

**Comment by xiaohu2015** [comment]
ok, I understand. Can you give some cases as a comparison?

**Comment by KevinChen880723** [comment]
@xiaohu2015 
Sure! Here is an example.
The first image is the input of IP-Adapter, the second one is a result from diffusers, and the third image is a result from Automatic1111. In my opinion, I think the middle one is pretty good, but the right one isn't that similar to the input.
![image](https://github.com/tencent-ailab/IP-Adapter/assets/55487740/df00fd49-49c5-4b7e-9305-a55809f7ddd4)

**Comment by xiaohu2015** [comment]
@KevinChen880723 they are so different？I think you should make sure that you should use the same configuration (e.g. sd model, sampler, seed) for diffusers and automatic1111
